thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -95,6 +95,7 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
     size32_t fixedEstSize;
     Owned<IRowWriter> pipewr;
     Owned<ISmartRowBuffer> piperd;
+    mutable CriticalSection critPiperd;
```

Comment:
random general comment on naming:
piperdCrit is a more natural name than critPiperd (and more consistent with other variables e.g. fixedEstSize)  - because piperd is an adjective/qualifier for the critical section, and in english qualifiers come before the noun.  Not worth changing in this PR.